users/vieting/experiments/switchboard/transducer/feat/experiments.py

```diff
@@ -277,6 +283,182 @@ def run_rasr_gt_stage1():
+    # the version above.
+    returnn_args_ctc_align["extra_args"]["chunking"] = (
+        {"classes": 64, "data": 64 * 4 * 80 + 200 - 80},
+        {"classes": 32, "data": 32 * 4 * 80},
+    )
+    feature_args = {
+        "class": "ScfNetwork",
+        "size_tf": 256 // 2,
+        "stride_tf": 10 // 2,
+        "preemphasis": 0.97,
+        "wave_norm": True,
+        "wave_cast": True,
+    }
+    common_args = {
+        "feature_args": feature_args,
+        "lr_args": {"dynamic_learning_rate": dynamic_learning_rate_scf},
+    }
+
+    nn_args, report_args_collection = get_nn_args_baseline(
+        nn_base_args={
+            "bs15k_v1_align-ctc-conf-e400": dict(
+                returnn_args=returnn_args_ctc_align,
+                report_args={"alignment": "ctc-conf-e401"},
+                lr_args={"dynamic_learning_rate": dynamic_learning_rate_scf},
+                feature_args=feature_args,
+            ),
+        },
+        num_epochs=300,
+        evaluation_epochs=[270, 280, 290, 300],
+        prefix="viterbi_scf80_",
+    )
+    config = copy.deepcopy(nn_args.returnn_recognition_configs["viterbi_scf80_bs15k_v1_align-ctc-conf-e400"].config)
+    config["extern_data"]["data"]["dtype"] = "float32"
+    config["extern_data"]["classes"]["dtype"] = "int32"
+    nn_args.returnn_recognition_configs["viterbi_scf80_bs15k_v1_align-ctc-conf-e400"].config = config
+    nn_system, report = run_nn_args(nn_args, report_args_collection, dev_corpora["transducer"])
+    return nn_system, report
+
+
+def run_scf_stage1_from_checkpoint():
+    ctc_alignment = get_ctc_alignment(use_scf_aligment=True)
+
+    gs.ALIAS_AND_OUTPUT_SUBDIR = "experiments/switchboard/transducer/feat/"
+    _, dev_corpora, _ = get_switchboard_data()
+    returnn_datasets = get_returnn_datasets_transducer_viterbi(context_window={"classes": 1, "data": 121})
+    returnn_datasets_hash_break = get_returnn_datasets_transducer_viterbi(
+        context_window={"classes": 1, "data": 121},
+        keep_hashes=False,
+    )
+    returnn_datasets_align_ctc = get_returnn_datasets_transducer_viterbi(
+        alignment=ctc_alignment,
+        features="waveform_pcm",
+    )
+    returnn_args = {
+        "batch_size": 15000,
+        "datasets": returnn_datasets,
+        "extra_args": {
+            # data sequence is longer by factor 4 because of subsampling and 80 because of feature extraction vs.
+            # raw waveform
+            "chunking": ({"classes": 64, "data": 64 * 4 * 80}, {"classes": 32, "data": 32 * 4 * 80}),
+            "gradient_clip": 20.0,
+            "learning_rate_control_error_measure": "sum_dev_score",
+            "min_learning_rate": 1e-6,
+            "preload_from_files": {
+                "existing-model": {
+                    "filename": "/u/maximilian.kannen/setups/20230406_feat/work/i6_core/returnn/training/ReturnnTrainingJob.O9Y8K5i3P1Qo/output/models/epoch.400.index",
```

Comment:
"filename":  tk.Path("/u/maximilian.kannen/setups/20230406_feat/alias/experiments/switchboard/ctc/feat/train_nn/conformer_bs2x5k_scf_baseline_preemphasis97_/output/models/epoch.400.index"),
 
 Like this?

Reply:
This is not as bad as a string absolute path, but better would be to get it from within the recipes, similar to how the alignment is obtained.

Reply:
So no hard-coded link into your setup. If you change your alias this will break. Or, even worse, if someone else wants to reproduce this, it is not possible.